_posts/2017-01-30-docker-frontend-tools.md

```diff
@@ -0,0 +1,146 @@
+---
+
+## Overview
+
+We at Savas Labs have been using Docker for local development and CI environments for some time. Recently we wanted to test out [Phase 2's Pattern Lab Starter 8 theme](https://github.com/phase2/pattern-lab-starter). However, to use this theme locally we needed to build a new Docker image for running applications that the theme depends on. In this post, I'll share:
+
+- A `Dockerfile` used to build an image with Node, npm, PHP, and Composer installed
+- A `docker-compose.yml` configuration and Docker commands for running theme commands such as `npm start` from within the container
+
+Along the way, I'll also provide:
+
+- A quick overview of why we use Docker for local development
+- Tips for building custom images and running common front-end applications inside containers.
+
+## Background
+
+We at Savas switched to using Docker for local development last year and we love it - so much so that we even proposed [a Drupalcon session on our approach and experience](https://events.drupal.org/node/17250). Using Docker makes it incredibly easy for developers to quickly spin up consistent local development environments that match production. In the past we used Vagrant and virtual machines for these purposes, but we've found Docker to be much faster when switching between multiple projects.
+
+Usually we build our Docker images from scratch to perfectly match production environments. However, for agile development and rapid prototyping, we sometimes make use of public Docker images. In these cases we've relied on [Wodby's Docker4Drupal project](https://github.com/wodby/docker4drupal), which is "a set of docker containers optimized for Drupal."
+
+We plan to post a more in-depth series of posts covering Docker for local development and CI environments in the near future, so be on the look out for that!
+
+We're also fans of using style guides and atomic design to drive front-end development. So when we heard of the [Pattern Lab Starter Drupal theme](https://github.com/phase2/pattern-lab-starter) which leverages [Pattern Lab](http://patternlab.io/) (a tool for building pattern-driven user interfaces using atomic design), we were very excited to try it out.
+
+## Building the Docker image
+
+To try out the Pattern Lab Starter theme we started with a fresh local Drupal 8 installation, and then quickly spun up our local Docker development environment [using Docker4Drupal](http://docs.docker4drupal.org/en/latest/#usage). We then copied the [Pattern Lab Starter code](https://github.com/phase2/pattern-lab-starter) to a new `custom/theme/patter_lab_starter` directory in our Drupal project.
+
+However, running the Phase 2 Pattern Lab Starter theme requires Node, npm, PHP, and Composer - the former for the node dependencies (Gulp, Bower, etc.) and the latter for running Pattern Lab.
```

Comment:
The use of "former" and "latter" is a bit confusing here, since I think of those terms as drawing a distinction between _2 things_, but here there are 4. I think you're saying
- "former" = Node
-  "latter" = npm, PHP, and Composer

Could you make it a bit more clear?